replicators/src/mysql_connector/connector.rs

```diff
@@ -128,8 +132,31 @@ impl MySqlBinlogConnector {
         };
 
         connector.register_as_replica().await?;
-        connector.request_binlog().await?;
-
+        let binlog_request = connector.request_binlog().await;
+        match binlog_request {
+            Ok(()) => (),
+            Err(mysql_async::Error::Server(ref err))
+                if mysql_srv::ErrorKind::from(err.code)
+                    == mysql_srv::ErrorKind::ER_MASTER_FATAL_ERROR_READING_BINLOG =>
+            {
+                // Requested binlog is not available, we need to re-snapshot
+                error!("Failed to request binlog: {}", err);
```

Comment:
Usually prefer structured logging for fields over string interpolation:

```suggestion
                error!(error = %err, "Failed to request binlog");
```